### DISCUSSION TITLE
why does xwayland try to ptrace the compositor and other clients?

### DISCUSSION THREAD

**Discussion by odomingao** [hyprwm/Hyprland#6902] [Q&A]
I'm trying to write AppArmor profiles for Hyprland and other hypr utilities, and I've noticed that xwayland attempts to ptrace everything, including the compositor. Is this supposed to happen? What is the purpose of this behavior? Restricting it doesn't seem to result in loss of functionality as far as I can tell, but I'm curious about it.

```
ALLOWED xwayland ptrace comm=Xwayland requested_mask=read denied_mask=read peer=hyprland
ALLOWED xwayland ptrace comm=Xwayland requested_mask=read denied_mask=read peer=xhost
ALLOWED xwayland ptrace comm=Xwayland requested_mask=read denied_mask=read peer=pavucontrol
```

**Comment by vaxerski** [comment]
no clue, ask xwayland devs

**Reply by odomingao** [reply]
Thanks. It must be specific to wlroots then, if it's not a Hyprland thing. This doesn't seem to be a thing in neither gnome nor kde
